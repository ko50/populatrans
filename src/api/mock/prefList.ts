import { Prefecture } from "models/prefecture";

export async function getMockedPrefectureList(): Promise<Prefecture[]> {
    const prefList: Prefecture[] = [
        {
            code: 1,
            name: "茨城県",
        },
        {
            code: 2,
            name: "埼玉県",
        },
        {
            code: 3,
            name: "東京都",
        },
    ];

    return prefList;
}
