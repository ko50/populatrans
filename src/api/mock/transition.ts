import { Prefecture } from "models/prefecture";
import { TransitionList } from "models/transition";

export async function getMockedPopulationTransition(
    prefList: Prefecture[]
): Promise<TransitionList> {
    const transition: TransitionList = {
        boundaryYear: 2020,
        transitions: [
            {
                pref: { code: 1, name: "茨城県" },
                populations: [
                    {
                        year: 2018,
                        value: 2000,
                    },
                    {
                        year: 2019,
                        value: 4000,
                    },
                    {
                        year: 2020,
                        value: 6000,
                    },
                    {
                        year: 2021,
                        value: 8000,
                    },
                ],
            },
            {
                pref: { code: 2, name: "東京都" },
                populations: [
                    {
                        year: 2018,
                        value: 1000,
                    },
                    {
                        year: 2019,
                        value: 2000,
                    },
                    {
                        year: 2020,
                        value: 3000,
                    },
                    {
                        year: 2021,
                        value: 4000,
                    },
                ],
            },
        ],
    };

    return transition;
}
