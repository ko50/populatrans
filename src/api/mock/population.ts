import { Transition } from "models/transition";

export async function getMockedPopulationComposition(): Promise<Transition[]> {
    const prefList: Transition[] = [
        {
            boundaryYear: 2020,
            populations: [
                {
                    year: 2018,
                    value: 2000,
                },
                {
                    year: 2019,
                    value: 4000,
                },
                {
                    year: 2020,
                    value: 6000,
                },
                {
                    year: 2021,
                    value: 8000,
                },
            ],
        },
        {
            boundaryYear: 2020,
            populations: [
                {
                    year: 2018,
                    value: 1000,
                },
                {
                    year: 2019,
                    value: 2000,
                },
                {
                    year: 2020,
                    value: 3000,
                },
                {
                    year: 2021,
                    value: 4000,
                },
            ],
        },
    ];

    return prefList;
}
