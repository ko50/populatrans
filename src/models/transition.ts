import { Population } from "models/population";

export type Transition = {
    boundaryYear: number;
    populations: Population[];
};
