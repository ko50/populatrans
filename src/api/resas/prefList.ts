import { resasClient } from "api/resas/config/client";
import { RESAS_BASE_PATH } from "api/resas/config/constants";
import { Response } from "api/resas/config/types";
import { Prefecture } from "models/prefecture";

type ResponseData = {
    prefCode: number;
    prefName: string;
};

export async function getResasPrefectureList(): Promise<Prefecture[]> {
    const url = RESAS_BASE_PATH + "/prefectures";
    const res = await resasClient.get<Response<ResponseData[]>>(url);

    const result = res.data.result;
    const prefectures = result.map<Prefecture>((data) => {
        return {
            code: data.prefCode,
            name: data.prefName,
        };
    });

    return prefectures;
}
