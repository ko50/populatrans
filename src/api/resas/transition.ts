import { resasClient } from "api/resas/config/client";
import { RESAS_BASE_PATH } from "api/resas/config/constants";
import { Response } from "api/resas/config/types";
import { Prefecture } from "models/prefecture";
import { Transition, TransitionList } from "models/transition";

type ResponseData = {
    boundaryYear: number;
    data: [
        // 総人口のみ考慮
        {
            label: "総人口";
            data: {
                year: number;
                value: number;
            }[];
        }
    ];
};

export async function getResasPopulationTransition(
    selected: Prefecture[]
): Promise<TransitionList> {
    const data: Transition[] = [];
    let boundaryYear: number;

    for (const p of selected) {
        await (async () => {
            const url =
                RESAS_BASE_PATH +
                "/population/composition/perYear?prefCode=" +
                p.code +
                "&cityCode=-";
            const _res = await resasClient.get<Response<ResponseData>>(url);

            const result = _res.data.result;
            const datum: Transition = {
                pref: p.name,
                populations: result.data[0].data,
            };

            boundaryYear = result.boundaryYear;
            data.push(datum);
        })();
    }

    const transitionList: TransitionList = {
        boundaryYear: boundaryYear!,
        transitions: data,
    };

    return transitionList;
}
