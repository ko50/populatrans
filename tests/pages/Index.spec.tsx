import React from "react";
import renderer from "react-test-renderer";
import Index from "pages/index";
import { getMockedPrefectureList } from "api/mock/prefList";
import { getMockedPopulationTransition } from "api/mock/transition";

describe("common home page", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Index
          prefList={[]}
          api={{
            prefListAPI: getMockedPrefectureList,
            populationTransitionAPI: getMockedPopulationTransition,
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
