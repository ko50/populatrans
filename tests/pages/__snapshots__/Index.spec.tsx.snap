// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common home page should render correctly 1`] = `
<div>
  <main
    className="main"
  >
    <div
      className="header"
    >
      POPULATrans
    </div>
    <div
      className="wrapper"
    >
      <div
        className="graph"
      >
        <div
          className="h1"
        >
          <h1
            className="text"
          >
            人口の推移
          </h1>
          <div
            className="underline"
          />
        </div>
        <div
          className="wrapper"
        >
          Loading
        </div>
      </div>
      <div
        className="prefForm"
      >
        <div
          className="h1"
        >
          <h1
            className="text"
          >
            都道府県
          </h1>
          <div
            className="underline"
          />
        </div>
        <div
          className="wrapper"
        />
      </div>
    </div>
  </main>
</div>
`;
